core/tests/test-dependencies/route-to-module-mapping-generator.ts

```diff
@@ -0,0 +1,403 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview This script is used to generate an Angular route to module mapping.
+ */
+
+import path from 'path';
+import {Route} from '@angular/router';
+import {
+  CallExpression,
+  ObjectLiteralExpression,
+  SourceFile,
+  ts,
+} from 'ts-morph';
+import {
+  project,
+  ROOT_DIRECTORY,
+  getDecorationNodesByTextFromSourceFile,
+  resolveModuleRelativeToRoot,
+  AngularDecorators,
+  getValueFromLiteralStringOrBinaryExpression,
+} from './typescript-ast-utilities';
+
+// List of routing module file paths.
+const ROUTING_MODULE_FILE_PATHS = [
+  path.resolve(
+    ROOT_DIRECTORY,
+    'core/templates/pages/oppia-root/routing/app.routing.module.ts'
+  ),
+  path.resolve(
+    ROOT_DIRECTORY,
+    'core/templates/pages/lightweight-oppia-root/routing/app.routing.module.ts'
+  ),
+];
+
+// List of routes that are not defined in routing modules.
+const MANUAL_ROUTE_TO_MODULE_MAPPING: Map<Route, string> = new Map([
```

Comment:
If the files in this list get removed will the script fail?

Reply:
Added a validation for that, Thanks.